### ISSUE TITLE
Improve exception safety with smart pointers

### ISSUE THREAD

**Issue by elfring** [smtrat/smtrat#20]
Would you like to wrap any pointer data members with the class template “[std::unique_ptr](https://en.wikipedia.org/wiki/Smart_pointer#unique_ptr "Description for the usage of smart pointers")”?

Update candidates:
- [Bound](https://github.com/smtrat/smtrat/blob/2d1c12d71f5e6c49c37f42d0ad9917c08c7fb963/src/lib/datastructures/VariableBounds.tpp#L12)
- [VSModule](https://github.com/smtrat/smtrat/blob/d2dfcfa21510fc103594b163eebd6262d5fdad60/src/smtrat-modules/VSModule/VSModule.cpp#L21)

**Comment by nafur** [comment]
Yes, modernizing with std::unique_ptr is one of the things I'd like to do (and have already done in other places), but this is simply a matter of time. I've not done this for those classes yet...
If you provide me with a pull request I shall happily accept it ;-)
